### PR TITLE
Bump @nextcloud/vue from 1.5.0 to 2.2.1

### PR DESCRIPTION
**Pull request by georgehrke**
Signed-off-by: Georg Ehrke <developer@georgehrke.com>

Breaking changes:
- [x] `#app-content` to `.app-content`
- [x] `#app-navigation` to `.app-navigation`
- [x] `#app-sidebar` to `.app-sidebar`
- [x]  `#content` to `.content`
- [x] Navigation List Slot
- [x] AppNavigationSettings Footer Slot

### CODE REVIEW

File: apps/settings/src/views/Users.vue [hunk 1]
```diff
@@ -92,58 +92,60 @@
...
-					<input id="sendWelcomeMail"
-						v-model="sendWelcomeMail"
-						:disabled="loadingSendMail"
-						type="checkbox"
-						class="checkbox">
-					<label for="sendWelcomeMail">{{ t('settings', 'Send email to new user') }}</label>
-				</div>
-			</AppNavigationSettings>
+			</template>
+			<template #footer>
```
Comment (skjnldsv) [inline apps/settings/src/views/Users.vue:96]: Wait, why did we need a footer slot? I completely missed this! :thinking:

**Review by skjnldsv — APPROVED** [review]
Thanks for this @georgehrke :heart:

File: apps/settings/src/views/Users.vue (hunk 1 above)
Comment (georgehrke) [reply apps/settings/src/views/Users.vue:96]: https://github.com/nextcloud/nextcloud-vue/releases/tag/v2.2.0 😉

Comment (skjnldsv) [reply apps/settings/src/views/Users.vue:96]: Interesting choice, I would have put the default slot to be the list and not the Navigation `new` button :shrug:

**Comment by skjnldsv** [comment]
Needs big rebase.

**Comment by georgehrke** [comment]
Rebased

**Comment by ChristophWurst** [comment]
> dependageorg

Nice

**Review by ChristophWurst — APPROVED** [review]
Not sure if ready to review but the code looks fine :)

**Comment by skjnldsv** [comment]
Acceptance apps/users fails & jsunit too.

**Comment by georgehrke** [comment]
```
PhantomJS 2.1.1 (Linux 0.0.0) DEBUG: 'OCA.Files.Settings initialized'
1011	PhantomJS 2.1.1 (Linux 0.0.0) ERROR
1012	  {
1013	    "message": "SyntaxError: Unexpected token 'const'\nat apps/files/js/dist/personal-settings.js:1:0",
1014	    "str": "SyntaxError: Unexpected token 'const'\nat apps/files/js/dist/personal-settings.js:1:0"
1015	  }
1016	PhantomJS 2.1.1 (Linux 0.0.0): Executed 0 of 0 ERROR (0.374 secs / 0 secs)
1017	PhantomJS 2.1.1 (Linux 0.0.0) ERROR
1018	  {
1019	    "message": "SyntaxError: Unexpected token 'const'\nat apps/files/js/dist/sidebar.js:1:0",
1020	    "str": "SyntaxError: Unexpected token 'const'\nat apps/files/js/dist/sidebar.js:1:0"
1021	  }
```

**Comment by skjnldsv** [comment]
I guess this is the same issue I reported recently.
There is an issue loading the scripts on files on master :thinking:

**Comment by georgehrke** [comment]
> I guess this is the same issue I reported recently.
There is an issue loading the scripts on files on master 🤔

The nc/vue bundle seems to be using const in the built version. I added it to the webpack config.

**Comment by skjnldsv** [comment]
> Transpile @nc/vue and some of its dependencies 2e6775b7e807609b531c66030b9c9b7e13be4b6b

I'm not sure abou tthis one! Don't we already transpile on the library?
https://github.com/nextcloud/nextcloud-vue/blob/master/babel.config.js

**Comment by danxuliu** [comment]
I have fixed the share link acceptance tests (basically the share menu was covering in some cases the _Copy link_ button; it should not happen with up to date browsers).

However, while checking this I have found a small issue (latest Firefox):

Share-Link-Menu-Out-Of-Bounds

Since https://github.com/nextcloud/nextcloud-vue/commit/f3b223a54ca5c2b9dc5acb7c131c61d698ba7aaa the Actions menu is always shown to the top if there is enough room (before this was done only for centered menus, so it did not apply to the share menu due to being aligned to the right). However, as can be seen in this case there is not enough room, and yet the menu is still shown to the top (and a scroll bar is not shown either). I do not know if the problem is in the Vue component itself or in the way that it is used in the sharing tab.

**Comment by skjnldsv** [comment]
> the problem is in the Vue component itself

this ^

**Comment by georgehrke** [comment]
> > the problem is in the Vue component itself
> 
> this ^

That seems to be an edge case with certain view heights.
@skjnldsv @rullzer Are you fine with merging this for now?

**Comment by rullzer** [comment]
> > > the problem is in the Vue component itself
> > 
> > 
> > this ^
> 
> That seems to be an edge case with certain view heights.
> @skjnldsv @rullzer Are you fine with merging this for now?

Yep lets do this. But be sure to create an issue so we don't lose track

**Comment by georgehrke** [comment]
Created an issue: https://github.com/nextcloud/nextcloud-vue/issues/1215

Let me rebase

**Pull request merged**